cmake/NSIS.template.in

```diff
@@ -539,20 +539,18 @@ FunctionEnd
 ; Component sections
 @CPACK_NSIS_COMPONENT_SECTIONS@
 
+!include "FileAssociation.nsh"
+
 Section "Open STL files with Cura"
-    WriteRegStr HKCR .stl "" "STL Model File"
-    DeleteRegValue HKCR .stl "Content Type"
-    WriteRegStr HKCR "Cura STL model file\DefaultIcon" "" "$INSTDIR\Cura.exe,0"
-    WriteRegStr HKCR "Cura STL model file\shell" "" "open"
-    WriteRegStr HKCR "Cura STL model file\shell\open\command" "" '"$INSTDIR\Cura.exe" "%1"'
+    ${RegisterExtension} "$INSTDIR\Cura.exe" ".stl" "Cura.model.STL"
 SectionEnd
 
 Section /o "Open OBJ files with Cura"
-    WriteRegStr HKCR .obj "" "Wavefront OBJ Model File"
-    DeleteRegValue HKCR .obj "Content Type"
-    WriteRegStr HKCR "Cura OBJ model file\DefaultIcon" "" "$INSTDIR\Cura.exe,0"
-    WriteRegStr HKCR "Cura OBJ model file\shell" "" "open"
-    WriteRegStr HKCR "Cura OBJ model file\shell\open\command" "" '"$INSTDIR\Cura.exe" "%1"'
+    ${RegisterExtension} "$INSTDIR\Cura.exe" ".obj" "Cura.model.OBJ"
+SectionEnd
+
+Section /o "Open 3MF files with Cura"
+    ${RegisterExtension} "$INSTDIR\Cura.exe" ".3mf" "Cura.model.3MF"
 SectionEnd
 
```

Comment:
NSIS script is pretty limited and lacks decent control structures. The only thing I can think of is some trickery with the jump instructions. But I'd personally stay away from that.